Integrations/integration-McAfee-ATD.yml

```diff
@@ -36,687 +37,665 @@ configuration:
-            return val;
-        }
+    HEARTBEATHEADERS = {
+        'Accept': 'application/vnd.ve.v1.0+json',
+        'Content-Type': 'application/json'
     }
 
-    function getSessionCredentials(){
-        res = sendRequest('/php/session.php','GET', LOGINHEADERS);
-        if (res) {
-            return res.results;
-        }
-    }
-
-    function getHeaders(){
-        var sess = getSessionCredentials();
-        return {'Accept': ['application/vnd.ve.v1.0+json'],
-                'VE-SDK-API':[Base64.encode(sess.session + ':' + sess.userId)]
-        };
-    }
-
-    function sendRequestRaw(uri, method, headers, body){
-        var res = http(
-                BASE+uri,
-                {
-                    Method: method,
-                    Headers: headers,
-                    Body: body
-                },
-                UNSECURE,
-                PROXY
-            );
-        if (res.StatusCode < 200 || res.StatusCode >= 300) {
-            throw 'Failed to use uri ' + BASE+uri + ', request status code: ' + res.StatusCode + ' and Body: ' + res.Body + '.';
+    ''' HELPERS '''
+    def stringArrToIntArr(stringArr):
+        ''' Converts a string array to int array '''
+        intArr = []
+        if isinstance(stringArr, list):
+            for index in range(len(stringArr)):
+                if stringArr[index]:
+                    intArr.append(int(stringArr[index]))
+        else:
+            intArr = [int(stringArr)]
+        return intArr
+
+    def castToInt(val):
+        ''' Casts a string to int '''
+        if isinstance(val, str):
+            return int(val)
+        else:
+            return val
+
+    def getSessionCredentials():
+        res = sendRequest('php/session.php','get', LOGINHEADERS)
+        if (res):
+            return res['results']
+
+    def getHeaders():
+        sess = getSessionCredentials()
+        return {
+            'Accept': 'application/vnd.ve.v1.0+json',
+            'VE-SDK-API': base64.b64encode(sess['session'] + ':' + sess['userId'])
         }
 
-        var uriStr = String(uri);
-        if (uriStr.indexOf("session") === -1) {
-            logout();
+    def sendRequestRaw(uri, method, headers = {}, body = {}, params={}, files={}):
+        ''' Makes an API call with the supplied uri, method, headers, body and params '''
+        url = '%s/%s' % (BASE, uri)
+        res = requests.request(
+            method,
+            url,
+            headers=headers,
+            data=body,
+            verify=USE_SSL,
+            params=params,
+            files=files
+        )
+        if res.status_code < 200 or res.status_code >= 300:
+                return_error('Got status code ' + str(res.status_code) + ' with body ' + res.content + ' with headers ' + str(res.headers))
+        uriStr = str(uri)
+        if (uriStr.find("session") == -1):
+            logout()
+        return res
+
+    def sendRequest(uri, method, headers = {}, body = {}, params={}, files={}):
```

Comment:
call it `http_request` and unite sendRequest and sendRequestRaw

Reply:
done